src/live.rs

```diff
@@ -187,440 +160,450 @@ pub fn pxe_ignition_unwrap(config: &PxeIgnitionUnwrapConfig) -> Result<()> {
+                    region.offset,
+                    cursor
+                );
+            }
+            copy_exactly_n(input, writer, region.offset - cursor, &mut buf)
+                .with_context(|| format!("copying bytes from {} to {}", cursor, region.offset))?;
+            if region.length != region.contents.len() {
+                bail!(
+                    "expected region contents length {}, found {}",
+                    region.length,
+                    region.contents.len()
+                );
+            }
+            writer.write_all(&region.contents).with_context(|| {
+                format!(
+                    "writing region for {} at offset {}",
+                    region.length, region.offset
+                )
+            })?;
+            cursor = input
+                .seek(SeekFrom::Current(region.length as i64))
+                .with_context(|| format!("seeking region length {}", region.length))?;
         }
 
         // write the remainder
         let mut write_buf = BufWriter::with_capacity(BUFFER_SIZE, writer);
         copy(
-            &mut BufReader::with_capacity(BUFFER_SIZE, &mut self.file),
+            &mut BufReader::with_capacity(BUFFER_SIZE, input),
             &mut write_buf,
         )
         .context("copying file")?;
         write_buf.flush().context("flushing output")?;
         Ok(())
     }
-
-    fn write_kargs(&mut self, kargs: &str) -> Result<()> {
-        let new_area = self.format_embed_area(kargs)?;
-
-        for offset in &self.kargs_offsets {
-            self.file
-                .seek(SeekFrom::Start(*offset))
-                .with_context(|| format!("seeking to karg area offset {}", *offset))?;
-            self.file
-                .write_all(&new_area)
-                .with_context(|| format!("writing karg embed area at offset {}", *offset))?;
-        }
-        Ok(())
-    }
 }
 
-// This is purposely not an impl function because we need to be able to call it while borrowing
-// parts of the struct (e.g. when iterating over the offsets).
-fn get_kargs_at_offset(file: &mut File, area_length: usize, offset: u64) -> Result<String> {
-    file.seek(SeekFrom::Start(offset))
-        .with_context(|| format!("seeking to karg area offset {}", offset))?;
-    let area = {
-        let mut buf = vec![0u8; area_length];
-        file.read_exact(&mut buf)
-            .with_context(|| format!("reading karg area at offset {}", offset))?;
-        String::from_utf8(buf).context("invalid UTF-8 in karg area")?
-    };
-    Ok(area.trim_end_matches('#').trim().into())
-}
+#[derive(Serialize)]
+struct KargEmbedAreas {
+    #[serde(rename = "default_kargs")]
+    default_region: Region,
+    #[serde(skip_serializing)]
+    default_args: String,
 
-enum ContentFile {
-    ForStdout(File),
-    InPlace(File),
-    Copied((NamedTempFile, PathBuf)),
+    #[serde(rename = "kargs")]
+    regions: Vec<Region>,
```

Comment:
And this one just `kargs`?

Reply:
I feel like it's clearer to be explicit that these fields have regions of unprocessed bytes (and in this case, multiple copies of same), rather than kargs per se.

Reply:
WDYT about splitting the difference and calling them `default_kargs_region` and `kargs_regions`? And then we can drop the serde renames.

Reply:
Could do, though the renames don't bother me that much.  The serde renames are for `--header`, which has a different perspective than the code does; `region` is an implementation detail there.